File: tests/python/test_identifiers.py
import pytest

from mlsnpdr.identifiers import make_module_uid, parse_module_uid


def test_module_uid_normalizes_legacy_spelling():
    assert (
        make_module_uid("PhysicalPPIN", "louvain", "c3", "m10")
        == "physicalppin__louvain__C3__M10"
    )


def test_module_uid_round_trip():
    identity = parse_module_uid("physicalppin__louvain__C3__M10")
    assert identity.legacy_module_id == "physicalPPIN|Louvain|C3|M10"
    assert identity.uid == "physicalppin__louvain__C3__M10"


def test_invalid_network_fails():
    with pytest.raises(ValueError, match="Unknown network"):
        make_module_uid("unknown", "WF", "C1", "M1")

File: tests/testthat/test-identifiers.R
test_that("module UID normalizes legacy spelling", {
  expect_identical(
    make_module_uid("PhysicalPPIN", "louvain", "c3", "m10"),
    "physicalppin__louvain__C3__M10"
  )
  expect_identical(
    make_module_uid("String", "WF", "C1", "M2"),
    "string__wf__C1__M2"
  )
})

test_that("module UID round-trips", {
  parsed <- parse_module_uid("physicalppin__louvain__C3__M10")
  expect_identical(parsed$legacy_module_id, "physicalPPIN|Louvain|C3|M10")
  expect_identical(parsed$module_uid, "physicalppin__louvain__C3__M10")
})

test_that("unknown identifiers fail loudly", {
  expect_error(make_module_uid("unknown", "WF", "C1", "M1"), "Unknown network")
  expect_error(make_module_uid("String", "other", "C1", "M1"), "Unknown module method")
})

